### PR TITLE
feat: Allow hmcts admin to classify uploaded documents as respondent statement (FPLA-2771)

### PR DESCRIPTION
**Pull request by tomaszpowroznik**
https://tools.hmcts.net/jira/browse/FPLA-2771

### CODE REVIEW

File: ccd-definition/ComplexTypes/CareSupervision/ManageDocuments/ManageDocumentType.json [hunk 1]
```diff
@@ -18,15 +18,6 @@
     "SecurityClassification": "Public",
     "FieldShowCondition": "type=\"DO NOT SHOW\""
   },
-  {
-    "LiveFrom": "01/01/2017",
-    "ID": "ManageDocumentType",
-    "ListElementCode": "relatedToHearing",
```
Comment (joez-f) [inline ccd-definition/ComplexTypes/CareSupervision/ManageDocuments/ManageDocumentType.json:24]: just doubling checking this complex type is used in transient field only, we are not storing this relatedToHearing in data?

File: e2e/steps_file.js [hunk 2]
```diff
@@ -332,6 +332,10 @@ module.exports = function () {
       return caseId;
     },
 
+    async grantCaseAccess(caseId, user, role) {
+      await caseHelper.grantCaseAccess(caseId, user.email, role);
+    },
+
```
Comment (joez-f) [inline e2e/steps_file.js:338]: is it needed for this PR? (minor, could be separated if not needed)

File: service/src/main/java/uk/gov/hmcts/reform/fpl/testingsupport/controllers/TestingSupportController.java [hunk 3]
```diff
@@ -80,4 +91,29 @@ public void populateCase(@PathVariable("caseId") Long caseId, @RequestBody Map<S
             throw e;
         }
     }
+
+    @PostMapping(value = "/testing-support/case/{caseId}/access")
+    public void grantCaseAccess(@PathVariable("caseId") Long caseId, @RequestBody Map<String, String> requestBody) {
```
Comment (joez-f) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/testingsupport/controllers/TestingSupportController.java:96]: same as comment on e2e, maybe can exist in separate PR if not relevant to respondent statement?

File: e2e/steps_file.js (hunk 2 above)
Comment (tomaszpowroznik) [reply e2e/steps_file.js:276]: yes, it is needed for this PR. 
in e2e i grant [SOLICITOR] role to user via api and then i check that this user cannot see confidential documents

File: service/src/main/java/uk/gov/hmcts/reform/fpl/testingsupport/controllers/TestingSupportController.java (hunk 3 above)
Comment (tomaszpowroznik) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/testingsupport/controllers/TestingSupportController.java:96]: same as above, this is used by added e2e test

File: e2e/steps_file.js (hunk 2 above)
Comment (tomaszpowroznik) [reply e2e/steps_file.js:276]: it is manageDocuments_test.js line 16

File: ccd-definition/ComplexTypes/CareSupervision/ManageDocuments/ManageDocumentType.json (hunk 1 above)
Comment (tomaszpowroznik) [reply ccd-definition/ComplexTypes/CareSupervision/ManageDocuments/ManageDocumentType.json:24]: this complex type is for transient data captured on 1st page of a flow. Now the fact that document is related to hearing is captured on 2nd page, thus does not belongs to this structure (ccd does not allow you to update same complex object on 2 pages). Case data has manageDocumentsRelatedToHearing which capture this information (it was added there by you  for LA part of story :) )

**Review by toireasam — DISMISSED** [review]
LGTM

File: ccd-definition/ComplexTypes/CareSupervision/ManageDocuments/ManageDocumentType.json (hunk 1 above)
Comment (joez-f) [reply ccd-definition/ComplexTypes/CareSupervision/ManageDocuments/ManageDocumentType.json:24]: thought so, just making sure :) did not want repeat of previous issue!

**Review by joez-f — APPROVED** [review]

**Review by joez-f — APPROVED** [review]

**Pull request merged**